core/org.eclipse.cdt.core/src/org/eclipse/cdt/core/build/CBuildConfiguration.java

```diff
@@ -208,6 +252,105 @@ public String getLaunchMode() {
 		return launchMode;
 	}
 
+	@Override
+	public ILaunchTarget getLaunchTarget() {
+		return launchTarget;
+	}
+
+	/**
+	 * @return the Launch Target associated with this {@link ICBuildConfiguration}
+	 * @throws CoreException
+	 *             if no Launch Target is found
+	 */
+	private ILaunchTarget getLaunchTargetFromProperties() throws CoreException {
+		String launchTargetId = getProperty(LAUNCH_TARGET_ID);
+		LaunchTargetInfo lti = getLaunchTargetInfo(launchTargetId);
+		if (lti != null) {
+			ILaunchTarget target = launchTargetManager.getLaunchTarget(lti.getTypeId(), lti.getId());
+			if (target != null) {
+				return target;
+			}
+		}
+		throw new CoreException(
+				Status.error(String.format("Error, Launch Target '%s' not found", launchTargetId), null)); //$NON-NLS-1$
+	}
+
+	/**
+	 *
+	 * @param target
+	 *            the Launch Target to associate with this {@link ICBuildConfiguration}
+	 * @throws CoreException
+	 *             if the Launch Target is null.
+	 */
+	private void saveLaunchTarget(ILaunchTarget target) throws CoreException {
+		if (target != null) {
+			launchTarget = target;
+			setProperty(LAUNCH_TARGET_ID, getLaunchTargetId(target));
+			return;
+		}
+		throw new CoreException(Status.error("Launch Target not set", null)); //$NON-NLS-1$
+	}
+
+	/**
+	 * @param launchTarget
+	 *            must not be null
+	 * @return a unique identifier for the launchTarget, composed of the {@link ILaunchTarget#getTypeId()} and
+	 *         {@link ILaunchTarget#getId()}
+	 * @since 9.0
+	 */
+	protected String getLaunchTargetId(ILaunchTarget launchTarget) {
+		String id = launchTarget.getId();
+		String typeId = launchTarget.getTypeId();
+		return typeId + LAUNCH_TARGET_ID_SEPARATOR + id;
+	}
+
+	/**
+	 * @since 9.0
+	 */
+	public class LaunchTargetInfo {
+		private String typeId;
+		private String id;
+
+		public LaunchTargetInfo(String typeId, String id) {
+			this.setTypeId(typeId);
+			this.setId(id);
+		}
+
+		public String getTypeId() {
+			return typeId;
+		}
+
+		public void setTypeId(String typeId) {
+			this.typeId = typeId;
+		}
+
+		public String getId() {
+			return id;
+		}
+
+		public void setId(String id) {
+			this.id = id;
+		}
+
+	}
+
+	/**
+	 * @param id
+	 * @return the info or null if the id is not recognised
+	 * @since 9.0
+	 */
+	protected LaunchTargetInfo getLaunchTargetInfo(String launchTargetId) {
+		if (launchTargetId != null) {
+			String[] split = launchTargetId.split(LAUNCH_TARGET_ID_SEPARATOR);
```

Comment:
Are IDs guaranteed to never have this separator by design? Instead of the joining and splitting, can you save the typeId to one field and the Id to another and then there is never ambiguity in serialization of parsing.